components/server/src/jobs/org-only-migration-job.ts

```diff
@@ -50,11 +58,11 @@ export class OrgOnlyMigrationJob implements Job<MigrationState> {
     }
 
     public async run(state?: MigrationState): Promise<MigrationState> {
```

Comment:
What feeds the state back into the migration? Given we select only ones which do not have Org ID, and then we set the Org ID, it seems we shouldn't really need the state at all, as the idea is we repeatedly reduce the size of the set we query such that eventually it has no results

Reply:
We query them in chunks of 3000 each time the job runs continuing with workspaces newer that the newest from the last run,

Reply:
But that's true irrespective of the "last time" right? Say we have 10k records to update, we take 3k, update them and on subsequent run, there are only 7k entries, out of which we take 3k again. That means we don't need the state for this

Reply:
Ah got it. You are right there should be no workspace older already. Do you insist on changing this (given it is a one-off migration)?